RecoTauTag/RecoTau/plugins/TauDiscriminationAgainstElectronDeadECAL.cc

```diff
@@ -0,0 +1,106 @@
+}
+// template specialiazation to get the correct (PF/PAT)Tau type names
+template <>
+std::string getTauTypeNameString<reco::PFTau>(bool capitalise) {
+  return capitalise ? "PFRecoTau" : "pfRecoTau";
+}
+template <>
+std::string getTauTypeNameString<pat::Tau>(bool capitalise) {
+  return capitalise ? "PATTau" : "patTau";
+}
+
+template <class TauType, class TauDiscriminator>
+class TauDiscriminationAgainstElectronDeadECALBase : public TauDiscriminationProducerBase<TauType, TauDiscriminator> {
+public:
+  typedef std::vector<TauType> TauCollection;
+  typedef edm::Ref<TauCollection> TauRef;
+  explicit TauDiscriminationAgainstElectronDeadECALBase(const edm::ParameterSet& cfg)
+      : TauDiscriminationProducerBase<TauType, TauDiscriminator>::TauDiscriminationProducerBase(cfg),
+        moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+        verbosity_(cfg.getParameter<int>("verbosity")),
+        antiElectronDeadECAL_(cfg) {}
+  ~TauDiscriminationAgainstElectronDeadECALBase() override {}
+
+  void beginEvent(const edm::Event& evt, const edm::EventSetup& es) override { antiElectronDeadECAL_.beginEvent(es); }
+
+  double discriminate(const TauRef& tau) const override {
+    if (verbosity_) {
+      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL")
+          << "<" + getTauTypeNameString<TauType>(true) + "AgainstElectronDeadECAL::discriminate>:";
+      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL") << " moduleLabel = " << moduleLabel_;
+      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL")
+          << " tau: Pt = " << tau->pt() << ", eta = " << tau->eta() << ", phi = " << tau->phi();
+    }
+    double discriminator = 1.;
+    if (antiElectronDeadECAL_(tau.get())) {
+      discriminator = 0.;
+    }
+    if (verbosity_) {
+      edm::LogPrint(getTauTypeNameString<TauType>(true) + "AgainstEleDeadECAL")
+          << "--> discriminator = " << discriminator;
+    }
+    return discriminator;
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  std::string moduleLabel_;
+  int verbosity_;
+
+  AntiElectronDeadECAL antiElectronDeadECAL_;
+};
+
+template <class TauType, class TauDiscriminator>
+void TauDiscriminationAgainstElectronDeadECALBase<TauType, TauDiscriminator>::fillDescriptions(
+    edm::ConfigurationDescriptions& descriptions) {
+  // {pfReco,pat}TauDiscriminationAgainstElectronDeadECAL
+  edm::ParameterSetDescription desc;
+
+  desc.add<double>("dR", 0.08);
+  desc.add<unsigned int>("minStatus", 12);
+  desc.add<bool>("extrapolateToECalEntrance", true);
+  desc.add<int>("verbosity", 0);
+
+  TauDiscriminationProducerBase<TauType, TauDiscriminator>::fillProducerDescriptions(
+      desc);  // inherited from the base-class
+
+  descriptions.add(getTauTypeNameString<TauType>(false) + "DiscriminationAgainstElectronDeadECAL", desc);  //base
```

Comment:
doesn't ```addWithDefaultLabel```  work here?
(I can imagine that the capitalization may be incorrect; but please check the default method if it works already.)

Reply:
We would like to keep uniform capitalisation over cfi's of tau modules. In addition, `std::string getTauTypeNameString(bool)` is used to provide TauType dependent string allowing to keep log printouts in coherent form with other tau modules (and previous version of this one). The latter is not critical, but nice feature.

Reply:
OK

Reply:
Checked that `addWithDefaultLabel` indeed works here as wanted, so adopted.